src/k8s-extension/azext_k8s_extension/partner_extensions/CostExport.py

```diff
@@ -0,0 +1,263 @@
+        )
+
+        extension = Extension(
+            extension_type=extension_type,
+            auto_upgrade_minor_version=auto_upgrade_minor_version,
+            release_train=release_train,
+            version=version,
+            scope=Scope(cluster=ScopeCluster(release_namespace=release_namespace), namespace=None),
+            configuration_settings=configuration_settings,
+            configuration_protected_settings=configuration_protected_settings,
+        )
+        create_identity = True
+        logger.info("deploying helm in cluster")
+        return extension, name, create_identity
+
+
+def _register_resource_provider(cmd, resource_provider):
+    if _is_resource_provider_registered(cmd, resource_provider):
+        logger.info("Resource provider '%s' is already registered.", resource_provider)
+        return
+    from azure.mgmt.resource.resources.models import ProviderRegistrationRequest, ProviderConsentDefinition
+
+    logger.warning("Registering resource provider %s ...", resource_provider)
+    properties = ProviderRegistrationRequest(
+        third_party_provider_consent=ProviderConsentDefinition(consent_to_authorization=True))
+
+    client = _providers_client_factory(cmd.cli_ctx)
+    try:
+        client.register(resource_provider, properties=properties)
+        # wait for registration to finish
+        timeout_secs = 120
+        registration = _is_resource_provider_registered(cmd, resource_provider)
+        start = time.time()
+        while not registration:
+            registration = _is_resource_provider_registered(cmd, resource_provider)
+            time.sleep(3)
+            if (time.time() - start) >= timeout_secs:
+                raise CLIInternalError(
+                    f"Timed out while waiting for the {resource_provider} resource provider to be registered.")
+    except Exception as e:
+        msg = ("This operation requires requires registering the resource provider {0}. "
+               "We were unable to perform that registration on your behalf: "
+               "Server responded with error message -- {1} . "
+               "Please check with your admin on permissions, "
+               "or try running registration manually with: az provider register --wait --namespace {0}")
+        raise ValidationError(resource_provider, msg.format(e.args)) from e
+    logger.info("Resource provider '%s' is now registered.", resource_provider)
+
+
+def _is_resource_provider_registered(cmd, resource_provider, subscription_id=None):
+    registered = None
+    if not subscription_id:
+        subscription_id = get_subscription_id(cmd.cli_ctx)
+    try:
+        providers_client = _providers_client_factory(cmd.cli_ctx, subscription_id)
+        registration_state = getattr(providers_client.get(resource_provider), 'registration_state', "NotRegistered")
+
+        registered = (registration_state and registration_state.lower() == 'registered')
```

Comment:
```suggestion
        registered = bool(registration_state and registration_state.lower() == 'registered')
```
Ensures that `registered` is always a bool - otherwise it'll be any falsy value that `registration_state` could contain (like `[]`, `{}`, `None`, `""`, `0`). Most likely not a problem, depending on how the result is used, but may as well make the type consistent here.